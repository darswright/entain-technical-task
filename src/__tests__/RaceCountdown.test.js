import { useContext } from "react";
import { render } from "@testing-library/react";
import RaceProvider, { RaceContext } from "./__mocks__/mockRaces.json";

const TestComponentWithAppContext = () => {
  const { error, categoriseRaces } = useContext(RaceContext);
  const races = categoriseRaces();
  if (error) return <div>error!</div>;
  return <div>Races: {races.length}</div>;
};

it("TestComponentWithAppContext", async () => {
  const { container } = render(
    <RaceProvider>
      <TestComponentWithAppContext />
    </RaceProvider>
  );

  expect(container).toMatchSnapshot();
});
