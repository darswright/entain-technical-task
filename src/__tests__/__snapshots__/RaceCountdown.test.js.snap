// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponentWithAppContext 1`] = `
<div>
  <div>
    Races: 
    10
  </div>
</div>
`;
